app/waf/src/controllers/adc.controller.ts

```diff
@@ -588,3 +580,132 @@ type AddonReqValues = {
   userToken: string;
   tenantId: string;
 };
+
+class AdcStateCtrlr {
+  private defaultCheck = (): Promise<boolean> => {
+    return Promise.resolve(true);
+  };
+  private states: AdcStateEntry[] = [
```

Comment:
Can we draw an ascii diagram about the state transition.....

Reply:
Yes, we need that. 
But it should be delay some time since doing it right now would burden us on the consistency between code and ASCII docs.

Reply:
It will require some effort to update doc when code changes. Nothing can help to accelerate that. Just hope to say a design spec might be required. A ascii doc might be better than a ppt, since it is easy for other persons to maintain and contribute.

Reply:
Maybe we can have a small review meeting for this pr, so that you can let other persons to know your design concept.

Reply:
Can I draft it as another PR, then reviewed with all f2f to populate this concept.?